crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -320,15 +348,15 @@ impl SettlementEncoder {
                 .token_index(self.order_trades[i].trade.order.data.buy_token)
                 .expect("missing buy token for existing trade");
         }
-        for i in 0..self.liquidity_order_trades.len() {
-            self.liquidity_order_trades[i].trade.sell_token_index = self
-                .token_index(self.liquidity_order_trades[i].trade.order.data.sell_token)
+        for i in 0..self.custom_price_trades.len() {
+            self.custom_price_trades[i].trade.sell_token_index = self
+                .token_index(self.custom_price_trades[i].trade.order.data.sell_token)
                 .expect("missing sell token for existing trade");
         }
     }
     fn modify_token_index_for_liquidity_orders_after_change(&mut self, offset: usize) {
-        for i in 0..self.liquidity_order_trades.len() {
-            self.liquidity_order_trades[i].buy_token_offset_index += offset;
+        for i in 0..self.custom_price_trades.len() {
+            self.custom_price_trades[i].buy_token_offset_index += offset;
         }
```

Comment:
Can't these be foreach loops?

Reply:
Didn't look to closely while renaming the variable because I wanted to change as little logic as possible.
I guess this particular change is small enough to be included anyway. 🤷‍♂️

Reply:
Added in [825ff25](https://github.com/cowprotocol/services/pull/712/commits/825ff2577b924e51914572c4bdf10d492a68bdf6).
The loops in the upper function can not be normal for loops because the loop borrows `self` mutably and the code inside the loop borrows `self` immutably.